light_curves/code_src/rubin_functions.py

```diff
@@ -0,0 +1,225 @@
+
+    Parameters
+    ----------
+    sample_table : astropy.table.Table
+        Table with columns ['coord', 'objectid', 'label'], where 'coord' is a SkyCoord.
+    rsp_tap : pyvo.dal.TAPService
+        Authenticated TAP service instance returned by `rubin_authenticate()`.
+    search_radius : float, deg
+        Radius of the cone search in degrees (default: 0.001).
+
+    Returns
+    -------
+    combined : astropy.table.Table
+        Combined table with columns ['coord_ra', 'coord_dec', 'objectId',
+        'in_objid', 'in_label'] tagging each DP0 Object match with the input sample info.
+    """
+    #could consider parallelizing this but for now we wait for TAP table upload and real data
+    all_tables = []
+    for row in sample_table:
+        ra = row['coord'].ra.deg
+        dec = row['coord'].dec.deg
+        in_objid = row['objectid']
+        in_label = row['label']
+        
+        query = textwrap.dedent(f"""
+        SELECT coord_ra, coord_dec, objectId
+        FROM dp02_dc2_catalogs.Object
+        WHERE CONTAINS(
+            POINT('ICRS', coord_ra, coord_dec),
+            CIRCLE('ICRS', {ra:.6f}, {dec:.6f}, {search_radius})
+        ) = 1
+        AND detect_isPrimary = 1
+        """).strip()
+        tbl = rsp_tap.run_sync(query).to_table()
+        # skip if no matches
+        if len(tbl) == 0:
+            continue
+        tbl['in_objid'] = in_objid
+        tbl['in_label'] = in_label
+        all_tables.append(tbl)
+
+    #in the case of an empty table, return an empty table
+    if not all_tables:
+        cols = ['coord_ra','coord_dec','objectId','in_objid','in_label']
+        # build a dict mapping each name to an empty list
+        empty_dict = {col: [] for col in cols}
+        return Table(empty_dict)
+
+    #otherwise, stack all the tables together
+    combined = vstack(all_tables)
+    return combined
+
+
+def rubin_access_lc_catalog(object_table, rsp_tap):
+    """
+    Retrieve forced-source light curves for given DP0 object IDs and attach sample metadata.
+
+    Parameters
+    ----------
+    object_table : astropy.table.Table
+        Table with ['objectId', 'in_objid', 'in_label'] from get_objectids().
+    rsp_tap : pyvo.dal.TAPService
+        Authenticated TAP service from rubin_authenticate().
+
+    Returns
+    -------
+    MultiIndexDFObject
+        Indexed by ['objectid', 'label', 'band', 'time'] with flux data.
+    """
+    #Pull out a unique, sorted list of integer IDs:
+    objids = sorted(set(object_table['objectId']))
+
+    #If there are no IDs, we don't need to search the light curve catalot
+    # and can just return an empty MultiIndexDFObject
+    if not objids:
+        # Create an empty MultiIndex 
+        mi = pd.MultiIndex.from_arrays(
+            [[], [], [], []],
+            names=['objectid', 'label', 'band', 'time']
+        )
+        empty_df = pd.DataFrame(index=mi)
+        return MultiIndexDFObject(empty_df)
```

Comment:
```suggestion
    if not objids:
        return MultiIndexDFObject()
```

No need for anything else.